lib/framework/scope.dart

```diff
@@ -151,11 +152,13 @@ class ScopeManager extends IsScopeManager with ViewBuilder, PageBindingManager {
     }
   }
 
-  /// create a copy of the parent's data scope
+  /// create a copy of the parent's data scope. the initialMap is generally coming from the imports
   @override
-  ScopeManager createChildScope({bool ephemeral = false}) {
+  ScopeManager createChildScope({bool ephemeral = false, List<ParsedCode>? importedCode}) {
     ScopeManager childScope =
-        ScopeManager(dataContext.clone(), pageData, ephemeral: ephemeral);
+        ScopeManager(dataContext
```

Comment:
So evalImports makes sense at the screen level as we want the screens to have separate context, but within a screen I would want my widgets, apis, item-templates,... to use the same code context. This is how the screen's Code block works today. createChildScope() is used for these purposes.

Reply:
I am not following you. A re-usable custom widget may have its own imports. So when we evaluate custom widget, we evaluate its imports as well.

Reply:
talked to Khurram. This is not what I understood. All good